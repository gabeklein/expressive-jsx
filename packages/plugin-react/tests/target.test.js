const test = require("./_adapter");

test("Simple modifier", `
  () => do {
    Foo: {
      color: red;
    }

    <Foo />
  };
`)

test("Attribute-based modifier", `
  () => do {
    Foo: {
      background: red;
    }

    hasWhiteText: {
      color: white;
    }

    <Foo hasWhiteText />
  };
`)

test("Nested modifier", `
  () => do {
    Foo: {
      color: red;
    }

    Bar: {
      Foo: {
        color: blue;
      }
    }

    <this>
      <Foo />
      <Bar>
        <Foo />
      </Bar>
    </this>
  };
`)

test("Explicitly use modifier", `
  () => do {
    Foo: {
      color: blue;
    }

    Baz: {
      use: Foo;
      color: red;
    }

    <this>
      <Baz />
      <Foo />
    </this>
  }
`)