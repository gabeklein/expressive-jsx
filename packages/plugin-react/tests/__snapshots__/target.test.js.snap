// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Attribute-based modifier 1`] = `
"() => create(Foo, {
  \\"className\\": \\"Foo_1u9\\"
});

Styles.include(\`
	.Foo_1u9 { 
		background: red;
		color: white;
	}
\`);"
`;

exports[`Attribute-based modifier 2`] = `
"() => <Foo className=\\"Foo_1u9\\" />;

Styles.include(\`
	.Foo_1u9 { 
		background: red;
		color: white;
	}
\`);"
`;

exports[`Explicitly use modifier 1`] = `
"() => create(Fragment, {}, create(Baz, {
  \\"className\\": \\"Foo_erd Baz_1u9\\"
}), create(Foo, {
  \\"className\\": \\"Foo_erd\\"
}));

Styles.include(\`
	.Foo_erd { 
		color: blue;
	}
	.Baz_1u9 { 
		color: red;
	}
\`);"
`;

exports[`Explicitly use modifier 2`] = `
"() => <Fragment><Baz className=\\"Foo_erd Baz_1u9\\" /><Foo className=\\"Foo_erd\\" /></Fragment>;

Styles.include(\`
	.Foo_erd { 
		color: blue;
	}
	.Baz_1u9 { 
		color: red;
	}
\`);"
`;

exports[`Nested modifier 1`] = `
"() => create(Fragment, {}, create(Foo, {
  \\"className\\": \\"Foo_erd\\"
}), create(Bar, {}, create(Foo, {
  \\"className\\": \\"Foo_erd Foo_51a\\"
})));

Styles.include(\`
	.Foo_erd { 
		color: red;
	}
	.Foo_51a { 
		color: blue;
	}
\`);"
`;

exports[`Nested modifier 2`] = `
"() => <Fragment><Foo className=\\"Foo_erd\\" /><Bar><Foo className=\\"Foo_erd Foo_51a\\" /></Bar></Fragment>;

Styles.include(\`
	.Foo_erd { 
		color: red;
	}
	.Foo_51a { 
		color: blue;
	}
\`);"
`;

exports[`Simple modifier 1`] = `
"() => create(Foo, {
  \\"className\\": \\"Foo_1u9\\"
});

Styles.include(\`
	.Foo_1u9 { 
		color: red;
	}
\`);"
`;

exports[`Simple modifier 2`] = `
"() => <Foo className=\\"Foo_1u9\\" />;

Styles.include(\`
	.Foo_1u9 { 
		color: red;
	}
\`);"
`;

exports[`collapse single-use modifiers 1`] = `
"() => create(Foo, {}, create(Bar, {
  \\"className\\": \\"Bar_7qc\\"
}));

Styles.include(\`
	.Bar_7qc { 
		border: red solid 1px;
		background: white;
		color: red;
	}
\`);"
`;

exports[`collapse single-use modifiers 2`] = `
"() => <Foo><Bar className=\\"Bar_7qc\\" /></Foo>;

Styles.include(\`
	.Bar_7qc { 
		border: red solid 1px;
		background: white;
		color: red;
	}
\`);"
`;
